### PR TITLE
MBS-9194: Add a bunch of sites the “other databases” whitelist

### PR DESCRIPTION
**Pull request by y-van-z**
For [Miku Hatsune Wiki](http://www5.atwiki.jp/hmiku/), [Baidu Baike](http://baike.baidu.com/), [Cancioneros Musicales Españoles](http://www.cancioneros.wiki), [Rock.com.ar](http://www.rock.com.ar), [MusicaPopular.cl](http://www.musicapopular.cl), [Bibliothèque nationale de France](http://catalogue.bnf.fr/), [VocaDB](http://vocadb.net/), [UtaiteDB](http://utaitedb.net/), [Irish Traditional Music Tune Index](https://www.irishtune.info/), [Cape Breton Fiddle Recording Index](http://www.cbfiddle.com/rx/), and [45worlds](http://www.45worlds.com):
1. Allow to enter them as `otherdatabases` relationship (with auto-selection, cleaning-up, and validation)
2. Display in the external links section of the sidebar

See [MBS-9194](https://tickets.metabrainz.org/browse/MBS-9194) and related tickets.

Additionally, it cleans up and validate [45cat](http://www.45cat.com) URLs which were already displayed in the sidebar.

### CODE REVIEW

**Comment by y-van-z** [comment]
Forced update adds:
- Improvements on matching, conversion and validation of BnF Catalogue URLs in https://github.com/metabrainz/musicbrainz-server/pull/437/commits/8d5a075deae8ae1a80e434255897dfe0cdf3c93d
  with updated tests in https://github.com/metabrainz/musicbrainz-server/pull/437/commits/6345ac7a41921a72ccc713bc347f772babc5688b
- Display of FRBNF number in the sidebar for BnF Catalogue URLs in https://github.com/metabrainz/musicbrainz-server/pull/437/commits/7c7e34ba427f824aa23d9ba4c16f4af1350eef61
- Display of world or classical artist role for 45worlds URLs in https://github.com/metabrainz/musicbrainz-server/pull/437/commits/b0e53cbcc55273da33d9b66b5ef52223bfa4879b

**Comment by y-van-z** [comment]
Forced update adds:
- Improvements on matching, conversion and validation of BnF Catalogue URLs in 454df6c330bf104a01cc866c33a94ab311814ef4
  with updated tests in 62270a5aacf231d3563040085c0a0e32dcc008fb
- Display of FRBNF number in the sidebar for BnF Catalogue URLs in b510bb76b0e900156c86f29c9989c8ae8b01c40a
- Display of world or classical artist role for 45worlds URLs in c11450f1ebfd4b732c09a47ab4115fa6bb16426e

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -1026,6 +1025,154 @@ const CLEANUPS = {
       return url;
     }
   },
+  a45cat: {
```
Comment (mwiencek) [inline root/static/scripts/edit/URLCleanup.js]: You can also have keys begin with numbers if you quote them. (Assuming that's what the "a" is for, though it hardly matters, just thought I'd mention it.)

**Comment by mwiencek** [comment]
Thank you!

**Pull request merged**